google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/StatementParserTest.java

```diff
@@ -1266,6 +1266,156 @@ public void testPostgreSQLGetQueryParameters() {
         parser.getQueryParameters("select '$2' from foo where bar=$1 and baz=$foo"));
   }
 
+  @Test
+  public void testGoogleSQLReturningClause() {
+    assumeTrue(dialect == Dialect.GOOGLE_STANDARD_SQL);
+
+    SpannerStatementParser parser = (SpannerStatementParser) this.parser;
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then\nreturn *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)\nthen\n\n\nreturn\n*"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then/*comment*/return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then return /*then return*/ *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then /*comment*/return(a)"))
```

Comment:
Could we also add some tests with (string) literals surrounding the `then return` clause?

So for example `delete from x where y='then return'` and `delete from x where y='z'then return *`

Reply:
Added.